### ISSUE TITLE
Arrays in Fontforge scripting broken

### ISSUE THREAD

**Issue by Font-Squirrel** [fontforge/fontforge#2523]
I think since the 8/24/2015 release, the arrays in Fontforge's native scripting system are broken. Specifically, the last member of any array is undefined, no matter what value you assigned to it. When printing an array, it looks something like [1,2,3,4,????]

A bunch of scripts broke when we upgraded to this release. Would love to have it resolved.

**Comment by jtanx** [comment]
To narrow the issue, may I ask what previous version was working?

**Comment by Font-Squirrel** [comment]
The latest version we had installed that worked properly was 20150430. This broke for us on both the Mac and Ubuntu installs.

**Comment by jtanx** [comment]
I can't seem to reproduce the issue. Running

``````c
a = Array(10)
i = 0
while ( i<10 )
   a[i] = i++
endloop

Print(a)
``````

It prints
~~~~~~
[0,1,2,3,4,5,6,7,8,9]
~~~~~~
as expected.

Do you have a sample script that causes this issue to occur?

**Comment by akovia** [comment]
This seems related to the problem I'm having, but mine is with StrSplit array results.
Hopefully fixed with ...
#2577 

**Example 1:** ($Familyname has a leading space)

````c
Print($familyname)
famsplit = (StrSplit($familyname," "))
Print(famsplit)
````

Prints

     In Stitches
    [,In,0]

**Example 2:** (Same code, different font with no leading space)

````
Disgusting Behavior
[Disgusting,<???>]
````

This is also effecting Print statements with quoted text.

````c
Print("famsplit = "+famsplit)
````

**returns**
Invalid type in integer expression




Arch Linux
fontforge 23:20 UTC 28-Sep-2015
libfontforge 20150928

**Comment by Font-Squirrel** [comment]
Perhaps it is the StrSplit() function causing issues. A sample script that does break:

```testarray = "a,b,c,d,e,f"

glyphs = StrSplit(testarray,",")

Print(glyphs)```

Results in:

`[a,b,c,d,e,0]`

**Comment by mskala** [comment]
I think the [a,b,c,d,e,0] issue is the bug fixed by #2577 .  If you're using a version prior to that, it'd be worth testing with a newer one.

**Comment by skef** [comment]
```
fontforge -quiet -lang=ff -script bg2523.ff 
PythonUI_Init()
copyUIMethodsToBaseTable()
[a,b,c,d,e,f]
```

Reading through this it seems like everything is either resolved or hangs on this problem. Will reopen if needed.
